clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -5852,6 +5852,17 @@ llvm::GlobalValue::LinkageTypes CodeGenModule::getLLVMLinkageForDeclarator(
   if (Linkage == GVA_AvailableExternally)
     return llvm::GlobalValue::AvailableExternallyLinkage;
 
+  // SYCL: Device code is not generally limited to one TU, but anything accessed
+  // from another translation unit is required to be annotated with the
+  // SYCL_EXTERNAL keyword. For any function or variable that does not have
+  // this, linkonce suffices. If -fno-sycl-rdc, though, we know there is only
+  // one translation unit and can so mark them internal instead.
```

Comment:
```suggestion
  // SYCL: Device code is not generally limited to one translation unit, but anything accessed
  // from another translation unit is required to be annotated with the
  // SYCL_EXTERNAL macro. For any function or variable that does not have
  // this, linkonce_odr suffices. If -fno-sycl-rdc is passed, we know there is only
  // one translation unit and can so mark functions internal.
```

Reply:
Mostly applied, thank you. The one thing is that the last "them" refers to both functions and variables, so replacing it with just "functions" does not cover it entirely. Are you okay with leaving that as "them", or would you prefer I repeat that as "functions and variables"?

Reply:
I'm ok with how it is applied now, thank you.